File: code/ch07/7.4.1_single_number.cpp
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        unordered_map<int, int> m;

        for (int i = 0; i < nums.size(); i++) {
            m[nums[i]]++;
        }

        for (auto it = m.begin(); it != m.end(); it++)
            if (it->second == 1)
                return it->first;

        return 0;
    }
};

File: code/ch07/7.4.1_single_number.java
class Solution {
    public int singleNumber(int[] nums) {
        Map<Integer, Integer> map = new HashMap<>();
        for (int num : nums)
            map.put(num, map.getOrDefault(num, 0) + 1);
        for (int num : map.keySet())
            if (map.get(num) == 1)
                return num;
        return 0;
    }
}